File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/usageFeeSetting/UsageFee.java
package org.taonaw.studio_reservation.domain.model.usageFeeSetting;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.taonaw.studio_reservation.domain.shared.Assertion;

@Getter
@EqualsAndHashCode
public class UsageFee {
    private final int amount;

    public UsageFee(int amount) {
        Assertion.argumentMin(amount, 0);
        this.amount = amount;
    }
}

File: src/backend/reservation/src/main/kotlin/org/taonaw/reservation/domain/model/usageFeeSetting/UsageFee.kt
package org.taonaw.reservation.domain.model.usageFeeSetting

import org.taonaw.reservation.domain.model.shared.DateTimeRange
import org.taonaw.reservation.domain.model.shared.Fee

data class UsageFee(
        private val usageFeeCondition: UsageFeeCondition,
        private val fee: Fee,
        private val type: Type) {

    enum class Type(private val value: Int) {
        HOURLY(1),
        FIXED(2);
    }

    fun getDateTimeRangeFee(calculationCondition: UsageFeeCalculationCondition): DateTimeRangeFee {
        return DateTimeRangeFee(usageFeeCondition.getMatchDateTimeRanges(calculationCondition), fee, type)
    }
}

data class DateTimeRangeFee(
        val dateTimeRanges: List<DateTimeRange>,
        val fee: Fee,
        val usageFeeType: UsageFee.Type) {

}